src/components/__test__/addMotorSlice.test.js

```diff
@@ -0,0 +1,38 @@
+import { configureStore } from '@reduxjs/toolkit';
+import addMotorSlice from '../../redux/slice/addMotorSlice'; // Replace 'yourSlice' with the correct path to your slice file
+
+describe('addMotorSlice', () => {
+  let store;
+
+  beforeEach(() => {
+    store = configureStore({
+      reducer: {
+        motor: addMotorSlice,
+      },
+    });
+  });
+
+  it('should have the correct initial state', () => {
+    const initialState = store.getState().motor;
+    expect(initialState).toEqual({
+      motors: [],
+      isLoading: false,
+      isError: null,
+    });
+  });
+
+  it('should handle fetchMotoData.fulfilled action', () => {
```

Comment:
Please implement the Jest snapshot test to component rather than slice because we already created a unit test for the backend. We need to test the user interface and render of the component in the fornt end
Jest snapshot testing is a feature provided by the Jest testing framework that allows you to capture and compare snapshots of the rendered output of your React components. This technique helps you detect unintended changes in the UI over time and makes it easier to identify regressions in your application's visual appearance. Its user interface test. for example`test('YourComponent snapshot test', () => {
  const { asFragment } = render(<YourComponent />);
  expect(asFragment()).toMatchSnapshot();
});
`